micrometer-core/src/main/java/io/micrometer/core/instrument/distribution/TimeWindowPercentileHistogram.java

```diff
@@ -37,10 +38,35 @@ public class TimeWindowPercentileHistogram extends AbstractTimeWindowHistogram<D
 
     private final DoubleHistogram intervalHistogram;
 
+    private final double[] histogramBuckets;
+
+    private final boolean isCumulativeBucketCounts;
+
     public TimeWindowPercentileHistogram(Clock clock, DistributionStatisticConfig distributionStatisticConfig,
             boolean supportsAggregablePercentiles) {
-        super(clock, distributionStatisticConfig, DoubleRecorder.class, supportsAggregablePercentiles);
+        this(clock, distributionStatisticConfig, supportsAggregablePercentiles, true, false);
+    }
+
+    public TimeWindowPercentileHistogram(Clock clock, DistributionStatisticConfig distributionStatisticConfig,
+            boolean supportsAggregablePercentiles, boolean isCumulativeBucketCounts, boolean includeInfinityBucket) {
```

Comment:
I didn't want to add this as a public constructor but see StackdriverHistogramUtil for why I couldn't do the same thing as TimeWindowFixedBoundaryHistogram (it's the `implementation` dependency on HdrHistogram).